### DISCUSSION TITLE
Is it sensible to use dnscrypt-proxy as a sinkhole adblocker?

### DISCUSSION THREAD

**Discussion by gyzerok** [DNSCrypt/dnscrypt-proxy#2384] [Q&A] [answered]
Hello everyone! I've been using your project for years now and love all the effort. Thank you!

Recently I figured I'd want to start using sinkhole approach for adblocking in my home network. I've discovered various projects to achieve this: PiHole, AdGuardHome and blocky. However I know that there is a blacklist feature in dnscrypt-proxy which seems like it could be used for the same purpose.

My question is: is it wise to use dnscrypt-proxy for that purpose? Maybe it's not designed to do it? Maybe performance will degrade too much if I use some huge 200k lines blacklists? Basically are there any significant trade-offs involved?

**Comment by jedisct1** [comment, marked as answer]
Yes, using `dnscrypt-proxy` for ad blocking (and other categories of content filtering such as malware, trackers, etc) is very common, and having 200k lines blacklist is fine.

In addition to exact matching, `dnscrypt-proxy` supports pattern matching, so very large lists are usually not necessary.

There's even a tool that can partially simplify lists for you, to avoid redundant rules: https://github.com/DNSCrypt/dnscrypt-proxy/tree/master/utils/generate-domains-blocklist

Also see:

* [filters](https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Filters)
* [combining blocklists](https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Combining-Blocklists)
* [public blocklists](https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Public-blocklist)

**Reply by gyzerok** [reply]
Great to hear, thank you!

Just a couple more questions if you don't mind:
1. Would it be ok to use such setup on an OpenWrt router performance-wise? It seems like other solutions are heavy on resources.
2. AdGuardHome can use dnscrypt-proxy. I just can't understand why people would use former in the first place if latter is enough to do the job. Is it just because of the UI?

**Comment by jedisct1** [comment]
1) Performance will be fine. But very large lists can take quite a bit of memory, so it mainly depends on how much memory the router has rather than its CPU.
2) Yes, a lot of people are not familiar with using a command line.

**Reply by gyzerok** [reply]
I see now. Appreciate your help!
